src/box/vinyl.c

```diff
@@ -2726,6 +2731,16 @@ vinyl_engine_free(struct engine *engine)
 	vy_env_delete(env);
 }
 
+/** Vinyl shutdown. Shutdown is stopping all internal fibers/threads. */
+static void
+vinyl_engine_shutdown(struct engine *engine)
+{
+	struct vy_env *env = vy_env(engine);
+	vy_scheduler_shutdown(&env->scheduler);
+	vy_squash_queue_shutdown(env->squash_queue);
+	vy_log_shutdown();
```

Comment:
Nit: Should be in `vy_env_shutdown`.

Reply:
Fixed.

```diff
diff --git a/src/box/vinyl.c b/src/box/vinyl.c
index 8058ba44b..44dc0160c 100644
--- a/src/box/vinyl.c
+++ b/src/box/vinyl.c
@@ -2731,6 +2731,14 @@ vinyl_engine_free(struct engine *engine)
        vy_env_delete(env);
 }

+static void
+vy_env_shutdown(struct vy_env *e)
+{
+       vy_scheduler_shutdown(&e->scheduler);
+       vy_squash_queue_shutdown(e->squash_queue);
+       vy_log_shutdown();
+}
+
 /**
  * Vinyl shutdown. Shutdown stops all internal fibers/threads.
  * It may yield.
@@ -2739,9 +2747,7 @@ static void
 vinyl_engine_shutdown(struct engine *engine)
 {
        struct vy_env *env = vy_env(engine);
-       vy_scheduler_shutdown(&env->scheduler);
-       vy_squash_queue_shutdown(env->squash_queue);
-       vy_log_shutdown();
+       vy_env_shutdown(env);
 }
```